spring-xd-module/src/main/java/org/springframework/xd/module/spark/streaming/SparkStreamingDriverModule.java

```diff
@@ -174,14 +186,50 @@ public void run() {
 	}
 
 	/**
-	 * Retrieve spark configuration properties from the {@link org.springframework.xd.spark.streaming.Processor} implementation.
+	 * Setup {@link org.apache.spark.SparkConf} for the given spark configuration properties.
+	 *
+	 * @param masterURL the spark cluster master URL
+	 * @param sparkConfigs the spark configuration properties
+	 * @return SparkConf for this spark streaming module
+	 */
+	private SparkConf setupSparkConf(String masterURL, Properties sparkConfigs) {
+		SparkConf sparkConf = new SparkConf()
+				// Set spark UI port to random available port to support multiple spark modules on the same host.
+				.set("spark.ui.port", String.valueOf(SocketUtils.findAvailableTcpPort()))
+				// Set the cores max so that multiple (at least a few) spark modules can be deployed on the same host.
+				.set("spark.cores.max", "3")
+				.setMaster(masterURL)
+				.setAppName(getDescriptor().getGroup() + "-" + getDescriptor().getModuleLabel());
```

Comment:
Yeah, it creates a unique App ID for them and I have just verified that as well.